src/ert/config/summary_config.py

```diff
@@ -97,6 +99,5 @@ def read_from_file(self, run_path: str, iens: int) -> xr.Dataset:
             coords={"time": axis, "name": keys},
         )
 
-
-def _should_load_summary_key(data_key, user_set_keys):
-    return any(fnmatch(data_key, key) for key in user_set_keys)
+    def _should_load_summary_key(self, data_key: Any, user_set_keys: set[str]) -> bool:
```

Comment:
it seems like based on the code i'm seeing in ecl for eclsum's [`__iter__`](https://github.com/equinor/ecl/blob/469546ebd1dbf5a56ad9390f47dfde65552114d3/python/ecl/summary/ecl_sum.py#L785) and [`keys`](https://github.com/equinor/ecl/blob/469546ebd1dbf5a56ad9390f47dfde65552114d3/python/ecl/summary/ecl_sum.py#L1513) methods, we can type the `data_key` as a string? although i'm a bit unsure what a `StringList` is :grimacing: is that the reason you used `Any` here? : D